capability/capability_linux.go

```diff
@@ -27,23 +28,31 @@ const (
 var (
 	capVers    uint32
 	capLastCap Cap
+	once sync.Once
 )
 
-func init() {
-	var hdr capHeader
-	capget(&hdr, nil)
-	capVers = hdr.version
 
-	if initLastCap() == nil {
-		CAP_LAST_CAP = capLastCap
-		if capLastCap > 31 {
-			capUpperMask = (uint32(1) << (uint(capLastCap) - 31)) - 1
-		} else {
-			capUpperMask = 0
+
+func GetCapMaskCapLast() (uint32,Cap) {
+	once.Do(func() {
+		var hdr capHeader
+		capget(&hdr, nil)
+		capVers = hdr.version
+	
+		if initLastCap() == nil {
+			CAP_LAST_CAP = capLastCap
```

Comment:
CAP_LAST_CAP is public.The external code can use it and after this change it will be broken.

Reply:
It is not compatible with old versions after my pr, do you have some idea?

Reply:
I try to solve this problem to make runc run faster .
https://github.com/opencontainers/runc/issues/4356